### ISSUE TITLE
Question: direct importing vs deconstruction

### ISSUE THREAD

**Issue by besh** [Semantic-Org/Semantic-UI-React#609]
Hey there. This isn't a feature request or a bug. Just a general question.

So, we have been doing `import { Button, Form, Dropdown } from 'stardust';` but that pulls in the entire stardust lib. The reality is, we're only using 10% of the stardust components so I started thinking about direct re-exporting in a lib file like so

``` js
export { default as Button } from 'stardust/dist/commonjs/elements/Button/Button';
```

Then when required, we can simply `import { Button, SomeOtherComponent } from 'lib/stardust'` and know that we're only bundling code that we're actually using.

So now I'm wondering, outside of the obvious breaking changes that may occur as stardust has updates, do you see any potential downsides to this approach?

**Comment by jeffcarbs** [comment]
Small custom builds is definitely something we're targeting, although that likely won't happen until we've shipped [v1.0.0](https://github.com/TechnologyAdvice/stardust/milestone/1). 

We're thinking something similar to https://github.com/lodash/babel-plugin-lodash which would allow you to write:

``` js
import { Button, Form, Dropdown } from 'stardust'
```

and have it automatically converted to:

``` js
import Button from 'stardust/elements/Button/Button'
import Form from 'stardust/collections/Form/Form'
import Dropdown from 'stardust/modules/Dropdown/Dropdown'
```

as a build step.

See here for more discussion on this: https://github.com/TechnologyAdvice/stardust/issues/524
See here for more insight into timelines: https://github.com/TechnologyAdvice/stardust/issues/606#issuecomment-251182192

Hope this helps!

**Comment by besh** [comment]
Great response. I'll hang out in the other issues for discussion.
